src/qinfer/derived_models.py

```diff
@@ -599,6 +602,220 @@ def update_timestep(self, modelparams, expparams):
         steps = steps.transpose((0, 2, 1))
         
         return modelparams[:, :, np.newaxis] + steps
+        
+class GaussianRandomWalkModel(DerivedModel):
+    r"""
+    Model such that after each time step, a random perturbation is 
+    added to each model parameter vector according to a 
+    zero-mean gaussian distribution.
+    
+    The :math:`n\times n` covariance matrix of this distribution is 
+    either fixed and known, or its entries are treated as unknown, 
+    being appended to the model parameters.
+    For diagonal covariance matrices, :math:`n` parameters are added to the model 
+    storing the square roots of the diagonal entries of the covariance matrix.
+    For dense covariance matrices, :math:`n(n+1)/2` parameters are added to 
+    the model, storing the entries of the lower triangular portion of the
+    Cholesky factorization of the covariance matrix.
+    
+    :param Model underlying_model: Model representing the likelihood with no
+        random walk added.
+    :param random_walk_idxs: A list of model parameter indeces to add the 
```

Comment:
In other places, we've used ``np.s_[:]`` to represent this idiom, but I do like the use of ``'all'`` as a synonym. In any case, the previous use of the ``s_`` idiom allows for using slicing notation as well as a fixed list, but it's definitely less familiar to users new to NumPy. Is there a way to perhaps have both benefits, do you think?